### ISSUE TITLE
It is acting strange under ReactOS

### ISSUE THREAD

**Issue by Zero3K** [weolar/xpchrome#8]
It keeps launching itself when ran under ReactOS. Here's a screenshot of the issue:

![VirtualBox_ReactOS_16_11_2023_13_07_48](https://github.com/weolar/xpchrome/assets/6194332/81cb2a63-9827-46ac-897a-c261fca3401c)

**Comment by weolar** [comment]
It looks like the UI is not being displayed?

**Comment by Zero3K** [comment]
Correct. It keeps launching itself until ReactOS crashes.

**Comment by Zero3K** [comment]
I wonder if it is a bug in your build of Chromium or a bug in ReactOS.

**Comment by weolar** [comment]
I am currently unable to confirm this, as I have not installed a ReactOS virtual machine to test this scenario.

**Comment by Zero3K** [comment]
Okay. I think it also affects Windows 2000 from what I have seen.

**Comment by weolar** [comment]
I don't think it will run on Win2000 because there are still a few APIs that I haven't implemented yet.

**Comment by Zero3K** [comment]
Would a debug log made while running it under ReactOS help?

**Comment by weolar** [comment]
I'm afraid it's not very useful. I need to find a virtual machine to debug it.

**Comment by Zero3K** [comment]
You can easily make one.

**Comment by Zero3K** [comment]
It looks to be a ReactOS bug according to the attached debug log.

[XPChrome_ReactOS.txt](https://github.com/weolar/xpchrome/files/13404517/XPChrome_ReactOS.txt)

**Comment by Zero3K** [comment]
Maybe someone could make a ticket regarding it on https://jira.reactos.org.

**Comment by Zero3K** [comment]
The new version doesn't give me the error dialog box, it just keeps loading itself with no UI showing.

**Comment by Zero3K** [comment]
Is there anything I can do to help?

**Comment by weolar** [comment]
Do you have a VMware image of ReactOS? This way I can run it directly for testing.

**Comment by Zero3K** [comment]
Not VirtualBox?

**Comment by Zero3K** [comment]
https://reactos.org/getbuilds/

**Comment by weolar** [comment]
ok, i try it

**Comment by Zero3K** [comment]
https://reactos.org/wiki/Debugging

**Comment by Zero3K** [comment]
Did you find anything?

**Comment by weolar** [comment]
Sorry, I've been quite busy these days and haven't had a chance to take a look yet.

**Comment by Zero3K** [comment]
I think making a ticket on https://jira.reactos.org would be a start just in case it is a ReactOS issue.

**Comment by MollySpacefield** [comment]
ZERO3K STOP HARASSING THIS GUY HOLY FUCK

**Comment by MollySpacefield** [comment]
Bro reacted 3 minutes ago, bro's sleep deprived

**Comment by Zero3K** [comment]
I won't comment in here any more until he responds with his findings.

**Comment by qwertyopus** [comment]
The same thing happens on Windows XP if WineD3D installed. I think it's because dxgi.dll. This file isn't in original Windows XP, but it is in ReactOS. May be replace dxgi.dll on something like axgi.dll with Hex Editor in chrome.dll allow you to solve the problem.

**Comment by Zero3K** [comment]
dxgi.dll isn't included by default in ReactOS.

**Comment by M18Bhoir** [comment]
I've noticed that Supermium crashes on ReactOS, and a similar issue occurs on Windows XP when WineD3D is installed, which also provides a dxgi.dll. This suggests the crash is likely due to an incompatibility with ReactOS's dxgi.dll implementation. As a test, modifying chrome.dll to bypass dxgi.dll might help confirm if this file is the source of the problem.
